src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs

```diff
@@ -37,7 +38,7 @@ internal sealed class JsonKeyValuePairConverter<TKey, TValue> : JsonConverter<Ke
                 ThrowHelper.ThrowJsonException();
             }
 
-            string propertyName = reader.GetString();
+            string? propertyName = reader.GetString();
```

Comment:
nit: Given we already check that the `reader.TokenType == JsonTokenType.PropertyName` above, we can use `!` here (and below).

```suggestion
            string propertyName = reader.GetString()!;
```

Reply:
By my understanding we only use `!` to suppress compiler warning where it is was not able/possible to evaluate all aspects. Preferred to use `?` where ever applicable because it shows compiler did right evaluation, '!' is more like suppressing compiler issue or so. Hope @stephentoub would give better explanation on it

Reply:
I believe @ahsonkhan is saying that GetString will never return null here because we already verified that the conditions in which it would return null aren't there.  True?  We often use `!` in such cases.

Reply:
Yes he meant that, ok then my understanding was wrong